xla/service/gpu/conv_algorithm_picker.cc

```diff
@@ -1111,12 +1111,14 @@ StatusOr<bool> GpuConvAlgorithmPicker::RunOnInstruction(HloInstruction* instr) {
       ShapeUtil::MakeShape(U8, {best_algo.scratch_bytes()}));
   Shape new_call_shape = ShapeUtil::MakeTupleShape(new_call_element_shapes);
 
-  TF_ASSIGN_OR_RETURN(CudnnConvBackendConfig backend_config,
-                      instr->backend_config<CudnnConvBackendConfig>());
+  TF_ASSIGN_OR_RETURN(GpuBackendConfig gpu_backend_config,
+                      instr->backend_config<GpuBackendConfig>());
+  CudnnConvBackendConfig backend_config =
+      gpu_backend_config.cudnn_conv_backend_config();
   *backend_config.mutable_algorithm() = best_algo.algorithm();
   backend_config.mutable_algorithm()->mutable_workspace_size()->set_value(
       best_algo.scratch_bytes());
-
+  *gpu_backend_config.mutable_cudnn_conv_backend_config() = backend_config;
```

Comment:
instead of making a deep copy, modifying then assigning back, you could just do

```
CudnnConvBackendConfig& backend_config = *gpu_backend_config.cudnn_conv_backend_config();
...
```
and delete the assignment by the end.